cmd/antrea-agent/options.go

```diff
@@ -121,27 +151,45 @@ func (o *Options) validate(args []string) error {
 		return fmt.Errorf("no positional arguments are supported")
 	}
 
-	if o.config.OVSDatapathType != string(ovsconfig.OVSDatapathSystem) {
-		return fmt.Errorf("OVS datapath type %s is not supported", o.config.OVSDatapathType)
-	}
-
-	if err := o.validateTLSOptions(); err != nil {
+	// Validate the Node type first, so we know what configs need to validate according to the Node type.
+	if err := o.validateNodeType(); err != nil {
 		return err
 	}
 
-	if config.ExternalNode.String() == o.config.NodeType && !features.DefaultFeatureGate.Enabled(features.ExternalNode) {
-		return fmt.Errorf("nodeType %s requires feature gate ExternalNode to be enabled", o.config.NodeType)
-	}
-
-	if o.config.NodeType == config.ExternalNode.String() {
-		o.nodeType = config.ExternalNode
-		return o.validateExternalNodeOptions()
-	} else if o.config.NodeType == config.K8sNode.String() {
-		o.nodeType = config.K8sNode
-		return o.validateK8sNodeOptions()
-	} else {
-		return fmt.Errorf("unsupported nodeType %s", o.config.NodeType)
+	// Common options which should be validated regardless of the Node type.
+	validators := []func() error{
+		o.validateTLSOptions,
+		o.validateOVSConfig,
+		o.validateAntreaPolicyConfig,
+	}
+	// Only validate configs applicable to the corresponding Node type.
+	if o.nodeType == config.K8sNode {
+		validators = append(validators,
```

Comment:
more of a long term idea, but if the dependencies between validators become a bit complex, we could imaging a better way to enforce correct ordering. For example:
```go
validators.add(vX, deps = [vY, vZ])
```
then we run a topological sort (at runtime) on the validators list to ensure we run them in the correct order.

For someone introducing a new feature / validator, it is easier and less error-prone to explicitly declare the dependencies and let the code handle the rest (as opposed to figuring out where to insert the new validator in a list).